src/test/ui/span/issue-24356.stderr

```diff
@@ -0,0 +1,10 @@
+error[E0046]: not all trait items implemented, missing: `Target`
+  --> $DIR/issue-24356.rs:30:9
+   |
+30 |         impl Deref for Thing {
+   |         ^ missing `Target` in implementation
+   |
+   = note: `Target` from trait: `type Target;`
```

Comment:
Duly noted, I didn't know they were frowned upon.

In order for this codepath to be executed the trait shouldn't have a span available, which as far as I know only happens when compiling against an extern crate. I hadn't noticed that [test/ui/cross-crate-macro-backtrace](https://github.com/estebank/rust/tree/40c2c0f833c0480c3cf3904fa3614cb0a01d7f87/src/test/ui/cross-crate-macro-backtrace) does exactly that. I'll change this to a ui test.